src/main/java/org/isf/medicalinventory/gui/InventoryWardBrowser.java

```diff
@@ -0,0 +1,636 @@
+import javax.swing.table.DefaultTableModel;
+
+import org.isf.generaldata.MessageBundle;
+import org.isf.medicalinventory.gui.InventoryWardEdit.InventoryListener;
+import org.isf.medicalinventory.manager.MedicalInventoryManager;
+import org.isf.medicalinventory.model.InventoryStatus;
+import org.isf.medicalinventory.model.InventoryType;
+import org.isf.medicalinventory.model.MedicalInventory;
+import org.isf.menu.manager.Context;
+import org.isf.utils.exception.OHServiceException;
+import org.isf.utils.exception.gui.OHServiceExceptionUtil;
+import org.isf.utils.jobjects.GoodDateChooser;
+import org.isf.utils.jobjects.MessageDialog;
+import org.isf.utils.jobjects.ModalJFrame;
+import org.isf.utils.time.TimeTools;
+import org.isf.ward.manager.WardBrowserManager;
+import org.isf.ward.model.Ward;
+import org.springframework.data.domain.Page;
+
+public class InventoryWardBrowser extends ModalJFrame implements InventoryListener {
+
+    private static final long serialVersionUID = 1L;
+
+    private GoodDateChooser jCalendarTo;
+    private GoodDateChooser jCalendarFrom;
+    private LocalDateTime dateFrom = TimeTools.getDateToday0();
+    private LocalDateTime dateTo = TimeTools.getDateToday24();
+    private JLabel jLabelTo;
+    private JLabel jLabelFrom;
+    private JPanel panelHeader;
+    private JPanel panelFooter;
+    private JPanel panelContent;
+    private JButton closeButton;
+    private JButton newButton;
+    private JButton editButton;
+    private JButton deleteButton;
+    private JButton viewButton;
+    private JScrollPane scrollPaneInventory;
+    private JTable jTableInventory;
+    private String[] pColums = {
+            MessageBundle.getMessage("angal.inventory.referenceshow.col").toUpperCase(),
+            MessageBundle.getMessage("angal.common.ward.col").toUpperCase(),
+            MessageBundle.getMessage("angal.common.date.col").toUpperCase(),
+            MessageBundle.getMessage("angal.inventory.status.col").toUpperCase(),
+            MessageBundle.getMessage("angal.common.user.col").toUpperCase()
+    };
+    private int[] pColumwidth = { 150, 150, 100, 100, 150 };
+    private JComboBox<String> statusComboBox;
+    private JLabel statusLabel;
+    JButton next;
+    JButton previous;
+    JComboBox<Integer> pagesCombo = new JComboBox<>();
+    JLabel under = new JLabel("/ 0 " + MessageBundle.getMessage("angal.inventory.page.text"));
+    private static int PAGE_SIZE = 50;
+    private int startIndex;
+    private int totalRows;
+    private MedicalInventoryManager medicalInventoryManager = Context.getApplicationContext().getBean(MedicalInventoryManager.class);
+    private WardBrowserManager wardBrowserManager = Context.getApplicationContext().getBean(WardBrowserManager.class);
+    private List<MedicalInventory> inventoryList;
+
+    public InventoryWardBrowser() {
+        initComponents();
+    }
+
+    private void initComponents() {
+
+        setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
```

Comment:
```suggestion
        setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
```
The import will be needed also